code/controllers/Processes/timer.dm

```diff
@@ -0,0 +1,77 @@
+var/global/datum/controller/process/timer/PStimer
+
+/datum/controller/process/timer
+	var/list/processing_timers = list()
+	var/list/hashes = list()
+
+/datum/controller/process/timer/setup()
+	name = "timer"
+	schedule_interval = 5 //every 0.5 seconds
+	PStimer = src
+
+/datum/controller/process/timer/statProcess()
+	..()
+	stat(null, "[processing_timers.len] timers")
+
+/datum/controller/process/timer/doWork()
+	if(!processing_timers.len)
+		disabled = 1 //nothing to do, lets stop firing.
+		return
+	for(var/datum/timedevent/event in processing_timers)
+		if(!event.thingToCall || qdeleted(event.thingToCall))
+			qdel(event)
+		if(event.timeToRun <= world.time)
+			spawn(-1)
```

Comment:
@MrStonedOne 

Are you absolutely positive about that? While I'm well acquainted with the problems `spawn` causes, performance wise, are you sure set watifor = 0 actually resolves the issue? I recall seeing one of your tests that bore out your claims to be true, but not matter how many times I profiled explosions with set waitfor = 0, they performed considerably worse with set watifor = 0 as opposed to `spawn`.

Is it possible that set waitfor = 0 has gains when dealing with large loops, but is worse when it's a single proc?